### ISSUE TITLE
QR Code scanning and action button beside spool to set spool ID via moonraker/ip

### ISSUE THREAD

**Issue by NonaSuomy** [Donkie/Spoolman#66]
OP Ref: https://github.com/Donkie/Spoolman/issues/10

I haven't tested if you can already do this with the edit button or not. Could you add an add button to the scan page where you can click add for a QR code that is not in your database? Which it will then fill in all the values for you from whatever the manufacturer sets. Say the producers of the filament wanted to support Spoolman and had premade Spoolman QR codes on their spools for instance that you just bought or a friend/reseller sent you a spool with the Spoolman QR tag on it.

Can you add another action button on the spool list that sets the spool id via a call to Moonraker?

**Comment by Donkie** [comment]
The current QR codes contains no data other than the spool ID. To do what you ask, it would have to be extended to support that, which could be done I guess, but is not what I had in mind from the beginning.

Spoolman has no knowledge of Moonraker's existence, which is how I want it to remain. The embedded client in Spoolman was never intended to be the *main* interface for Spoolman. The intention was always that the frontends (Mainsail etc) would do the actual client. They know about Moonraker, and can thus easily set the spool and so on.

**Comment by NonaSuomy** [comment]
>The current QR codes contains no data other than the spool ID. To do what you ask, it would have to be extended to support that, which could be done I guess, but is not what I had in mind from the beginning.

Makes sense. I think this should be extended to enable this before things get too popular. Would make spool entry leaps and bounds easier to catalog. 

I'm going to change the title to only this for a feature request then move the second part to another one.
